src/python/pants/build_graph/build_configuration.py

```diff
@@ -28,7 +28,7 @@
 
 # No goal or target_type can have a name from this set, so that `./pants help <name>`
 # is unambiguous.
-_RESERVED_NAMES = {"api-types", "global", "goals", "targets", "tools"}
+_RESERVED_NAMES = {"api-types", "global", "goals", "subsystems", "targets", "tools"}
```

Comment:
The `subsystems` name should've been reserved already, as there is a `./pants help subsystems` to list all subsystems.